### PR TITLE
feat: [#176205711] Allow the script to run multiple times

### PR DESCRIPTION
**Pull request by gquadrati**

### CODE REVIEW

File: mac [hunk 1]
```diff
@@ -254,15 +262,29 @@ fancy_echo "nodenv configured."
...
-ln -s "$ZSH_CUSTOM/themes/spaceship-prompt/spaceship.zsh-theme" "$ZSH_CUSTOM/themes/spaceship.zsh-theme"
-sed -i "" "/ZSH_THEME=*/ZSC_THEME=\"spaceship\"/g" .zshrc
+
+if [ ! -d "$SPACESHIP_ROOT" ]; then
+  git clone https://github.com/denysdovhan/spaceship-prompt.git "$SPACESHIP_ROOT"
+  ln -s "$SPACESHIP_ROOT/spaceship.zsh-theme" "$ZSH_CUSTOM/themes/spaceship.zsh-theme"
+  sed -i "" "s/^ZSH_THEME=.*/ZSH_THEME=\"spaceship\"/g" .zshrc
+else
+  git -C "$SPACESHIP_ROOT" pull
+fi
```
Comment (balanza) [inline mac]: It is not clear to me the intended behaviour. How is `$ZSH` set? How the user would decide to install a custom or a standard zsh?

_suggestion: you can add a block comment to explain the decision you made_

File: mac [hunk 2]
```diff
@@ -243,9 +242,18 @@ fancy_echo "patching nodenv init into .zshrc file"
...
-mkdir -p "$(nodenv root)/plugins"
-git clone https://github.com/pine/nodenv-yarn-install.git "$(nodenv root)/plugins/nodenv-yarn-install"
+
+if [ ! -d "$YARN_ROOT" ]; then
+  mkdir -p "$NODENV_ROOT_PLUGIN"
+  git clone https://github.com/pine/nodenv-yarn-install.git "$YARN_ROOT"
+else
+  git -C "$YARN_ROOT" pull
+fi
+
```
Comment (balanza) [inline mac:256]: What if `$YARN_ROOT`, for any reason, is not a git repository? Then `git -C "$YARN_ROOT" pull` will fail make the whole script crash. 
If you just want to update the plugin, then you could just remove the folder:

```suggestion
fancy_echo "patching yarn with nodenv"

NODENV_ROOT_PLUGIN="$(nodenv root)/plugins"
if [ ! -d "$NODENV_ROOT_PLUGIN" ]; then
  mkdir -p $NODENV_ROOT_PLUGIN
fi

NODENV_YARN_ROOT="$NODENV_ROOT_PLUGIN/nodenv-yarn-install"
# remove previous plugin installation 
if [ -d "$NODENV_YARN_ROOT" ]; then
  rm -r $NODENV_YARN_ROOT 2> /dev/null
fi

git clone --depth=1 https://github.com/pine/nodenv-yarn-install.git "$NODENV_YARN_ROOT"

```
Does it make sense to you?

File: mac (hunk 1 above)
Comment (gquadrati) [reply mac]: updated in the new commit. It now uses zsh to load ohmyzsh variables

File: mac (hunk 2 above)
Comment (gquadrati) [reply mac:256]: I personally don't like the idea to remove someone else's folder, especially if it doesn't contain the .git folder as we expect. I'd prefer to have an exception in this particular case

Comment (balanza) [reply mac:256]: it's a good point!

**Pull request closed without merging**